tests/unit/modules/zypper_test.py

```diff
@@ -414,37 +437,233 @@ def test_repo_value_info(self):
         with zypper_patcher:
             zypper.mod_repo(
-                'mock-repo-name',
+                name,
                 **{
-                    'disabled': False,
-                    'url': 'http://repo.url/some/path',
-                    'gpgkey': 'http://repo.key',
+                    'url': url,
+                    'gpgautoimport': True
+                }
+            )
+            self.assertEqual(
+                zypper.__zypper__.xml.call.call_args_list,
+                [
+                    call('--gpg-auto-import-keys', 'refresh', name)
+                ]
+            )
+            zypper.__zypper__.refreshable.xml.call.assert_not_called()
+
+    def test_add_repo_modify_repo_and_refresh_repo_with_gpgautoimport(self):
+        '''
+        Test mod_repo adds the new repo,
+        modifies the new repo when refresh is True and
+        refreshes the repo with
+            `zypper --gpg-auto-import-keys refresh <repo-name>`
+
+        :return:
+        '''
+        zypper_patcher = patch.multiple(
+            'salt.modules.zypper', **self.zypper_patcher_config)
+
+        url = self.new_repo_config['url']
+        name = self.new_repo_config['name']
+        with zypper_patcher:
+            zypper.mod_repo(
+                name,
+                **{
+                    'url': url,
+                    'refresh': True,
+                    'gpgautoimport': True
+                }
+            )
+            self.assertEqual(
+                zypper.__zypper__.xml.call.call_args_list,
+                [
+                    call('ar', url, name),
+                    call('--gpg-auto-import-keys', 'refresh', name)
+                ]
+            )
+            zypper.__zypper__.refreshable.xml.call.assert_called_once_with(
+                '--gpg-auto-import-keys', 'mr', '--refresh', name
+            )
+
+    def test_modify_existing_repo_and_refresh_repo_with_gpgautoimport(self):
+        '''
+        Test mod_repo detects the repo already exists,
+        modifies the existing repo when refresh is True and
+        refreshes the repo with
+            `zypper --gpg-auto-import-keys refresh <repo-name>`
+
+        :return:
+        '''
+        url = self.new_repo_config['url']
+        name = self.new_repo_config['name']
+        self.zypper_patcher_config['_get_configured_repos'] = Mock(
+            **{'return_value.sections.return_value': [name]}
+        )
+        zypper_patcher = patch.multiple(
+            'salt.modules.zypper', **self.zypper_patcher_config)
+
+        with zypper_patcher:
+            zypper.mod_repo(
+                name,
+                **{
+                    'url': url,
                     'refresh': True,
                     'gpgautoimport': True
                 }
             )
+            self.assertEqual(
+                zypper.__zypper__.xml.call.call_args_list,
+                [
+                    call('--gpg-auto-import-keys', 'refresh', name)
+                ]
```

Comment:
It doesn't have to be 5 codelines. 1890 U.S. Census is over 😉 and we're not at Hollerith's Tabulating Machine times, where 80 chars punch cards comes from. It can be just one line (and everywhere else).